osidb/mixins.py

```diff
@@ -229,15 +230,186 @@ def get_queryset(self):
+        """
+        acl = uuid.UUID(generate_acls([group])[0])
+        # store for printing
+        self.acl_group_map[acl] = group
+        return acl
+
+    @cached_property
+    def acls_public_read(self):
+        """
+        get set of public read ACLs
+        """
+        return {self.group2acl(group) for group in settings.PUBLIC_READ_GROUPS}
+
+    @cached_property
+    def acls_public_write(self):
+        """
+        get set of public write ACLs
+        """
+        return {self.group2acl(settings.PUBLIC_WRITE_GROUP)}
+
+    @cached_property
+    def acls_embargo_read(self):
+        """
+        get set of embargo read ACLs
+        """
+        return {self.group2acl(settings.EMBARGO_READ_GROUP)}
+
+    @cached_property
+    def acls_embargo_write(self):
+        """
+        get set of embargo write ACLs
+        """
+        return {self.group2acl(settings.EMBARGO_WRITE_GROUP)}
+
+    @cached_property
+    def acls_read(self):
+        """
+        get set of read ACLs
+        """
+        return self.acls_public_read | self.acls_embargo_read
+
+    @cached_property
+    def acls_write(self):
+        """
+        get set of write ACLs
+        """
+        return self.acls_public_write | self.acls_embargo_write
+
+    @cached_property
+    def acls_public(self):
+        """
+        get set of public ACLs
+        """
+        return self.acls_public_read | self.acls_public_write
+
+    @cached_property
+    def acls_embargo(self):
+        """
+        get set of embargo ACLs
+        """
+        return self.acls_embargo_read | self.acls_embargo_write
+
+    @cached_property
+    def acls_all(self):
+        """
+        get set of all ACLs
+        """
+        return self.acls_read | self.acls_write
+
+    def _validate_acls_known(self):
+        """
+        check that all the ACLs are known
+        """
+        for acl in self.acl_read + self.acl_write:
+            if acl not in self.acls_all:
+                groups = ", ".join([self.acl2group(acl) for acl in self.acls_all])
+                raise ValidationError(
+                    # here the printing of the actual group is problematic
+                    # as it is already a hash so we at least print all known
+                    f"Unknown ACL group given - known are: {groups}"
+                )
+
+    def _validate_acl_meaningful(self):
+        """
+        validate that the ACLs are set meaninfully in a way that read ACLs
+        contain read groups only and write ACLS only the write ones
+        """
+        for acl in self.acl_read:
+            if acl not in self.acls_read:
+                raise ValidationError(
+                    f"Read ACL contains non-read ACL group: {self.acl2group(acl)}"
+                )
+
+        for acl in self.acl_write:
+            if acl not in self.acls_write:
+                raise ValidationError(
+                    f"Write ACL contains non-write ACL group: {self.acl2group(acl)}"
```

Comment:
nitpick: I would probably do two different validations here

Reply:
Good point. I will spit it.